contracts/contracts/core/BidderRegistry.sol

```diff
@@ -406,21 +406,66 @@ contract BidderRegistry is
     /**
      * @dev Check the deposit of a bidder.
      * @param bidder The address of the bidder.
-     * @param window The window for which the deposit is being checked.
-     * @return The deposited amount for the bidder.
+     * @param provider The address of the provider.
+     * @return The available deposited amount for the bidder.
      */
     function getDeposit(
         address bidder,
-        uint256 window
+        address provider
+    ) external view returns (uint256) {
+        return deposits[bidder][provider].availableAmount;
+    }
+
+    function getDepositConsideringWithdrawalRequest(
+        address bidder,
+        address provider
+    ) external view returns (uint256) {
+        Deposit storage deposit = deposits[bidder][provider];
+        if (!deposit.exists || deposit.withdrawalRequestOccurrence.exists) {
+            return 0;
+        }
+        return deposit.availableAmount;
+    }
+
+    function getEscrowedAmount(
+        address bidder,
+        address provider
     ) external view returns (uint256) {
-        return lockedFunds[bidder][window];
+        return deposits[bidder][provider].escrowedAmount;
+    }
+
+    function withdrawalRequestExists(
+        address bidder,
+        address provider
+    ) external view returns (bool) {
+        return deposits[bidder][provider].withdrawalRequestOccurrence.exists;
     }
 
     /// @return protocolFee amount not yet transferred to recipient
     function getAccumulatedProtocolFee() external view returns (uint256) {
         return protocolFeeTracker.accumulatedAmount;
     }
 
+    function _depositAsBidder(address provider, uint256 amount) internal {
+        address bidder = msg.sender;
+        Deposit storage deposit = deposits[bidder][provider];
+        if (deposit.exists) {
+            require(!deposit.withdrawalRequestOccurrence.exists, 
+                WithdrawalOccurrenceExists(bidder, provider, deposit.withdrawalRequestOccurrence.timestamp));
+            deposit.availableAmount += amount;
+        } else {
+            deposits[bidder][provider] = Deposit({
+                exists: true,
+                availableAmount: amount,
+                escrowedAmount: 0,
+                withdrawalRequestOccurrence: TimestampOccurrence.Occurrence({
+                    exists: false,
+                    timestamp: 0})
+            });
+        }
+        emit BidderDeposited(bidder, provider, amount);
```

Comment:
Why not add the available amount in the event as well? That way we dont have to query all the events from the past. We can straight away use the latest event?